src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -467,7 +472,7 @@ void MergeTreeData::checkProperties(
                                 "Primary key must be a prefix of the sorting key, "
                                 "but the column in the position {} is {}", i, sorting_key_column +", not " + pk_column);
 
-            if (!primary_key_columns_set.emplace(pk_column).second)
+            if (!primary_key_columns_set.emplace(pk_column).second && !allow_suspicious_indices && !attach)
```

Comment:
Why do we check for `attach`? (just wondering)

Reply:
There may be a scenario that a table create in an old version of CH without setting `allow_suspicious_indices` and it used same expressions for index. After updating CH, it may got an exception and the table can't be attached.